R/setkey.R

```diff
@@ -332,6 +332,7 @@ CJ <- function(..., sorted = TRUE, unique = FALSE)
     dups = FALSE # fix for #1513
     # using rep.int instead of rep speeds things up considerably (but attributes are dropped).
     j = lapply(l, class)  # changed "vapply" to avoid errors with "ordered" "factor" input
+    tzones = lapply(l, function(col) attr(col, 'tzone'))
```

Comment:
Makes sense. Was trying to mirror the code that copies over classes. Any reason the `j = ...` line above my edit isn't inside the `else if`?